tests/queries/test_bulk_update.py

```diff
@@ -300,6 +300,33 @@ def test_json_field(self):
             JSONFieldNullable.objects.filter(json_field__has_key="c"), objs
         )
 
+    @skipUnlessDBFeature("supports_json_field")
+    def test_json_field_sql_null_value(self):
+        """
+        Setting a JSONField to None results in SQL NULL.
+        """
+        obj = JSONFieldNullable.objects.create(json_field={})
+        obj.json_field = None
+        JSONFieldNullable.objects.bulk_update([obj], fields=["json_field"])
+        obj.refresh_from_db()
+        sql_null_qs = JSONFieldNullable.objects.filter(json_field__isnull=True)
+        self.assertSequenceEqual(sql_null_qs, [obj])
+
+    @skipUnlessDBFeature("supports_json_field")
+    def test_json_field_json_null_value(self):
+        """
+        Setting a JSONField to Value(None, output_field=JSONField()) results
+        in a JSON null.
+        """
+        obj = JSONFieldNullable.objects.create(json_field={})
+        obj.json_field = Value(None, output_field=JSONField())
+        JSONFieldNullable.objects.bulk_update([obj], fields=["json_field"])
+        obj.refresh_from_db()
+        json_null_qs = JSONFieldNullable.objects.filter(
+            json_field=Value(None, output_field=JSONField())
+        )
+        self.assertSequenceEqual(json_null_qs, [obj])
+
```

Comment:
How about we write this as a single test with subTest?
```suggestion
    @skipUnlessDBFeature("supports_json_field")
    def test_json_field_null_value(self):
        """
        Setting a JSONField to null results in the proper null representation.
        """
        json_null = Value(None, output_field=JSONField())
        for null, lookups in [
            (None, {"json_field__isnull": True}),
            (json_null, {"json_field": json_null}),
        ]:
            obj = JSONFieldNullable.objects.create(json_field={})
            obj.json_field = null
            with self.subTest(null=null):
                result = JSONFieldNullable.objects.bulk_update([obj], fields=["json_field"])
                self.assertEqual(result, 1)
                obj.refresh_from_db()
                qs = JSONFieldNullable.objects.filter(**lookups)
                self.assertSequenceEqual(qs, [obj])

```

Reply:
Thanks @nessita and @charettes for the reviews.

I did a git bisect for 
```py
from .models import JSONFieldNullable
from django.test import TestCase, skipUnlessDBFeature


class Test(TestCase):
    @skipUnlessDBFeature("supports_json_field")
    def test_json_field_sql_null_value(self):
        """
        Setting a JSONField to None results in SQL NULL.
        """
        obj = JSONFieldNullable.objects.create(json_field={})
        obj.json_field = None
        JSONFieldNullable.objects.bulk_update([obj], fields=["json_field"])
        obj.refresh_from_db()
        sql_null_qs = JSONFieldNullable.objects.filter(json_field__isnull=True)
        self.assertSequenceEqual(sql_null_qs, [obj])

```
which is the issue reported in the ticket. the issue reported in the ticket and found the first bad commit to be  00c690efbc0b10f67924687f24a7b30397bf47d9 (main), 1df0f998ae8d1626adaa9b807a43e4cbdfc590e2 (5.2.x).

As for the tests that I added in this PR for JSON null, I had presumed that they were missing. I hadn't realised that the tests were also failing in previous Django versions as well.

SO it would seem that there are two separate issues here. The first is a regression in 00c690efbc0b10f67924687f24a7b30397bf47d9 and a release blocker, and the second is a normal bug. I'm not sure how to proceed here.

Do we percolate `for_save` (which as Simon and Adam correctly pointed out is the cause of the release blocker), and then fix explicit `JSON null` for `bulk_update()` later?